cmd/status-cli/message.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/common"
+	"github.com/status-im/status-go/protocol/protobuf"
+	"github.com/status-im/status-go/protocol/requests"
+
+	"github.com/urfave/cli/v2"
+)
+
+func sendContactRequest(cCtx *cli.Context, from *StatusCLI, toID string) error {
+	from.logger.Info("send contact request, contact public key: ", toID)
+	request := &requests.SendContactRequest{
+		ID:      toID,
+		Message: "Hello!",
+	}
+	resp, err := from.messenger.SendContactRequest(cCtx.Context, request)
+	from.logger.Info("function SendContactRequest response.messages: ", resp.Messages())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func sendContactRequestAcceptance(cCtx *cli.Context, from *StatusCLI, msgID string) error {
+	from.logger.Info("accept contact request, message ID: ", msgID)
+	resp, err := from.messenger.AcceptContactRequest(cCtx.Context, &requests.AcceptContactRequest{ID: types.Hex2Bytes(msgID)})
+	if err != nil {
+		return err
+	}
+	from.logger.Info("function AcceptContactRequest response: ", resp.Messages())
+
+	return nil
+}
+
+func sendDirectMessage(ctx context.Context, from *StatusCLI, text string) error {
+	if len(from.messenger.MutualContacts()) == 0 {
+		return nil
+	}
+	chat := from.messenger.Chat(from.messenger.MutualContacts()[0].ID)
+	from.logger.Info("send message to contact: ", chat.ID)
+
+	clock, timestamp := chat.NextClockAndTimestamp(from.messenger.GetTransport())
```

Comment:
I believe there should be a way to do this without making `GetTransport` a public method of Messenger.